src/context/AuthContext/provider.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useContext, useMemo } from 'react';
+import { AccountContext } from '~/context/AccountContext';
+import { EExternalIdentityStatus } from '~/context/AccountContext/constants';
+import { useLocalStorage } from '~/hooks/utility';
+import {
+  TConnectModalType,
+  TIdentityModalType,
+  REGEX_MOBILE_DEVICE,
+} from './constants';
+import AuthContext from './context';
+
+interface IAuthProviderProps {
+  children: React.ReactNode;
+}
+const AuthProvider = ({ children }: IAuthProviderProps) => {
+  const { externalIdentity, selectedAccount } = useContext(AccountContext);
+
+  const [showAuth, setShowAuth] = useLocalStorage('showAuth', true);
+
+  const [verified, setVerified] = useState(false);
+  const [connectPopup, setConnectPopup] = useState<TConnectModalType | null>(
+    null,
+  );
+  const [identityPopup, setIdentityPopup] = useState<TIdentityModalType | null>(
+    null,
+  );
+  const [isNewWalletMobile, setIsNewWalletMobile] = useState(false);
+
+  useEffect(() => {
+    if (externalIdentity?.status === EExternalIdentityStatus.VERIFIED) {
+      setVerified(true);
```

Comment:
rather than relying on the cddService we can just check if the key has an identity from the account context provider which reads directly from the connected chain.